internal/controllers/nmc_reconciler.go

```diff
@@ -565,94 +565,69 @@ func (h *nmcReconcilerHelperImpl) UpdateNodeLabelsAndRecordEvents(ctx context.Co
 	nodeModuleReadyLabels := sets.New[types.NamespacedName]()
 	deprecatedNodeModuleReadyLabels := sets.New[string]()
 
-	for label := range node.GetLabels() {
-		if ok, namespace, name := utils.IsKernelModuleReadyNodeLabel(label); ok {
-			nodeModuleReadyLabels.Insert(types.NamespacedName{Namespace: namespace, Name: name})
-		}
-
-		if utils.IsDeprecatedKernelModuleReadyNodeLabel(label) {
-			deprecatedNodeModuleReadyLabels.Insert(label)
-		}
-	}
+	getKernelModuleReadyLabels(&nodeModuleReadyLabels, &deprecatedNodeModuleReadyLabels, node)
```

Comment:
A `get*` function should return something. In your case, looks like a `set*` function will be a better fit.

BUT, I would keep it a `get*` function that return a new `Set` and then intersect the returned value with `nodeModuleReadyLabels`.

Also, I would split this function into 2, one that deals with labels and second one with deprecated labels.